pkg/chains/signing.go

```diff
@@ -57,22 +78,58 @@ type TaskRunSigner struct {
 	SecretPath        string
 }
 
-// IsSigned determines whether a TaskRun has already been signed.
-func IsSigned(tr *v1beta1.TaskRun) bool {
+// SignedState returns TaskRun state.
+func SignedState(tr *v1beta1.TaskRun) State {
```

Comment:
this function should return:

Passed: if `ChainsAnnotation=true`
Failed: if `ChainsAnnotation=failed`
Retrying: if `ChainsRetriesAnnotation<3`

in that order

Reply:
What happens when there is no `ChainsAnnotation`? Also, what is wrong with the present implementation?